contracts/contracts/GrantRoundManager.sol

```diff
@@ -0,0 +1,171 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+pragma solidity ^0.8.5;
+
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/utils/Address.sol";
+import "@uniswap/v3-periphery/contracts/interfaces/ISwapRouter.sol";
+import "./GrantRegistry.sol";
+import "./GrantRound.sol";
+
+contract GrantRoundManager {
+  using Address for address;
+  using SafeERC20 for IERC20;
+
+  /// @notice Donation Object
+  struct Donation {
+    uint96 grantId; // grant ID to which donation is being made
+    GrantRound[] rounds; // rounds against which the donation should be counted
+    IERC20 tokenIn; // token in which the user made the donation
+    uint24 fee; // selected fee tier
+    uint256 deadline; // deadline by when swap has to be happen
```

Comment:
nit: have the comment read: "unix timestamp that represents when the swap must be executed"

Reply:
Done in 02163ce